### PR TITLE
sdcc: Add version 4.1.0

### PR DESCRIPTION
**Pull request by HUMORCE**
Closes https://github.com/ScoopInstaller/Main/issues/1994

### CODE REVIEW

**Comment by phgmacedo** [comment]
@HUMORCE Do you think it would be a good idea to add the hash to the autoupdate? I'm always getting hash errors on scoop. 
```
 "autoupdate": {
        "architecture": {
            "64bit": {
                "url": "https://sourceforge.net/projects/sdcc/files/sdcc-win64/$version/sdcc-$version-x64-setup.exe#dl.7z",
                "hash": {
                    "url": "https://sourceforge.net/projects/sdcc/rss?path=/sdcc-win64",
                    "find": "\"md5\">(\\w+.)"
                }
            },
            "32bit": {
                "url": "https://sourceforge.net/projects/sdcc/files/sdcc-win32/$version/sdcc-$version-setup.exe#dl.7z",
                "hash": {
                    "url": "https://sourceforge.net/projects/sdcc/rss?path=/sdcc-win32",
                    "find": "\"md5\">(\\w+.)"
                }
            }
        }
    }
```

**Comment by HUMORCE** [comment]
/verify

@phgmacedo 
for sourceforge, the scoop has built-in methods to get the hash, and the manifest is works fine, now.
```powershell
$ .\bin\checkver.ps1 -App sdcc -U
sdcc: 4.1.0 (scoop version is test) autoupdate available
Autoupdating sdcc
Found: sha1:d186f8f4b0e97f53de56285f692c58863cf8d6a5 using Sourceforge Mode
Pinging https://sourceforge.net/projects/sdcc/files/sdcc-win32/4.1.0/sdcc-4.1.0-setup.exe#dl.7z to verify URL accessibility
Found: sha1:3193fbb5f77091a7bb26bd1fa45da111b65f087e using Sourceforge Mode
Pinging https://sourceforge.net/projects/sdcc/files/sdcc-win64/4.1.0/sdcc-4.1.0-x64-setup.exe#dl.7z to verify URL accessibility
Writing updated sdcc manifest
DONE
$ scoop install sdcc
Installing 'sdcc' (4.1.0) [64bit] [nuke]
Loading sdcc-4.1.0-x64-setup.exe from cache
Checking hash of sdcc-4.1.0-x64-setup.exe ... ok.
Extracting sdcc-4.1.0-x64-setup.exe#dl.7z ... done.
Linking ~\scoop\apps\sdcc\current => ~\scoop\apps\sdcc\4.1.0
Creating shim for 'sdcc'.
Running post-install script...
'sdcc' (4.1.0) was installed successfully!
```

File: bucket/sdcc.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+    "homepage": "http://sdcc.sourceforge.net",
+    "description": "A retargettable, optimizing Standard C (ANSI C89, ISO C99, ISO C11) compiler suite that targets the Intel MCS51 based, Maxim (Dallas) DS80C390 variants, Freescale (Motorola) HC08 based, Zilog Z80 based MCUs, Padauk and STMicroelectronics STM8.",
+    "version": "4.1.0",
+    "license": "GPL-2.0-only|GPL-3.0-only",
```
Comment (HUMORCE) [inline bucket/sdcc.json:5]: i'm not sure what licenses sdcc uses.

**Comment by phgmacedo** [comment]
From the manual, it is a collection of tools under multiple licenses:
* GPL2
* LGPL2
* GPL2
* xlib/libpng
* boost / BSL-1

File: bucket/sdcc.json (hunk 1 above)
Comment (phgmacedo) [reply bucket/sdcc.json:5]: From the manual, it is a collection of tools under multiple licenses: 
* GPL2 
* LGPL2 
* GPL3 
* xlib/libpng 
* boost / BSL-1

**Review by chawyehsu — APPROVED** [review]

**Pull request merged**